runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/MemoryPoolAssigner.java

```diff
@@ -47,18 +47,18 @@
   private final MemoryPool memoryPool;
 
   @Inject
-  public MemoryPoolAssigner(@Parameter(JobConf.MaxOffheapMb.class) final int maxOffheapMb,
+  public MemoryPoolAssigner(@Parameter(JobConf.ExecutorMemoryMb.class) final int memory,
+                            @Parameter(JobConf.MaxOffheapRatio.class) final double maxOffheapRatio,
                             @Parameter(JobConf.ChunkSizeKb.class) final int chunkSizeKb) {
+
+    int maxOffheapMb = (int) (memory * maxOffheapRatio);
     if (chunkSizeKb < MIN_CHUNK_SIZE_KB) {
       throw new IllegalArgumentException("Chunk size too small. Minimum chunk size is 4KB");
     }
     final long maxNumChunks = (long) maxOffheapMb * 1024 / chunkSizeKb;
     if (maxNumChunks > Integer.MAX_VALUE) {
       throw new IllegalArgumentException("Too many pages to allocate (exceeds MAX_INT)");
     }
-    if (maxNumChunks < 1) {
-      throw new IllegalArgumentException("The given amount of memory amounted to less than one chunk.");
-    }
```

Comment:
Let's put this back in, as we can assume that the max off heap ratio is not 0 at all times.

Reply:
I've added this condition, thanks